Sources/Signal.swift

```diff
@@ -1506,6 +1506,25 @@ extension Signal {
 	public func debounce(_ interval: TimeInterval, on scheduler: DateScheduler) -> Signal<Value, Error> {
 		return flatMapEvent(Signal.Event.debounce(interval, on: scheduler))
 	}
+
+	/// Forward the latest values on `scheduler` every `interval`.
+	///
+	/// - seealso: `throttle`
+	/// - seealso: `debounce`
+	///
+	/// - note: If the input signal terminates, the returned signal will
+	///         terminate immediately without forwarding the values
+	///	        currently being accumulated.
```

Comment:
In the spirit of #605, maybe there should be an option to forward the values first?

Reply:
Good point!

Reply:
After looking into this I can't find a way to make it work consistently, given:
When you complete the Signal when you'd expect is for it to still delay up to the supplied interval, and only then deliver the values followed immediately by a completed event. 
In the event of `interrupted` or `failed`, you'd need to forward those events immediately, so the values would also need to be delivered immediately. 
This would probably create tricky behaviour for a lot of people consuming this function, since you'd expect values to be delivered exactly `interval` apart (or a multiple of `interval` if you're ignoring when empty).

So should it only not discard in the case of `.completed`?